### DISCUSSION TITLE
Piper TTS as a windows integration

### DISCUSSION THREAD

**Discussion by SCBionicle** [rhasspy/piper#577] [Q&A]
Does anyone know of any Piper TTS based utility that allows you to install Piper as an SAPI5 or similar based application?

I have a bunch of games, and simulator plugins, that can use TTS generated voices for dialog but require them to use the windows speech synthesis API. I don't want to have to pay the data rates for Amazon Polly (I think) and Azure. Ultimately, I prefer something that works offline.

**Comment by Lej77** [comment]
I wrote a SAPI5 TTS engine in Rust that works with piper models, you can find it at <https://github.com/Lej77/windows-text-to-speech>. It is a bit bare bones but it seems to works well enough for my needs. Note that I could not find any way to make it work with the [`Windows.Media.SpeechSynthesis.SpeechSynthesizer`](https://learn.microsoft.com/en-us/uwp/api/windows.media.speechsynthesis.speechsynthesizer?view=winrt-26100) class and the `Remarks` on that page says:

> Only Microsoft-signed voices installed on the system can be used to generate speech.

So I suspect it is not easily fixed.
